client/checkout/utils/upe.js

```diff
@@ -279,12 +280,13 @@ export const getPaymentMethodTypes = ( paymentMethodType ) => {
 };
 
 /**
- * Returns the value of the email input on the blocks checkout page.
+ * Returns the email value from store API.
  *
- * @return {string} The value of email input.
+ * @return {string} The email value.
  */
 export const getBlocksEmailValue = () => {
-	return document.getElementById( 'email' ).value;
+	const cartData = wp.data.select( 'wc/store/cart' ).getCustomerData();
+	return cartData?.billingAddress?.email || '';
```

Comment:
I thought why not use store API instead of targeting the element?

Reply:
Hah, this doesn't make sense indeed. I moved back to getting the value from the DOM in 8be97f00555d977c6bbad0c3a142682cc520772d